### PR TITLE
delete volume placeholder when error

### PR DESCRIPTION
**Pull request by qieqieplus**
Delete placeholder from volumes when NewVolume()  failed

Also fix:
[store.go#L259](weed/storage/store.go#L259)

```
	found, err := location.deleteVolumeById(vid)
	if found {
		if err == nil {
			glog.V(0).Infof("volume %d is deleted", vid)
		} else {
			glog.V(0).Infof("delete volume %d: %v", vid, err)
		}
	}
```

won't log err, since `found` is always false when `deleteVolumeById` returns an error

### CODE REVIEW

File: weed/storage/disk_location.go [hunk 1]
```diff
@@ -207,16 +208,15 @@ func (l *DiskLocation) DeleteCollectionFromDiskLocation(collection string) (e er
...
-	v, ok := l.volumes[vid]
-	if !ok {
-		return
-	}
-	e = v.Destroy()
-	if e != nil {
-		return
+	v, found := l.volumes[vid]
+	if found {
+		if v != nil {
```
Comment (chrislusf) [inline weed/storage/disk_location.go:213]: Why put nil in the map in the previous PR?

**Review by chrislusf — CHANGES_REQUESTED** [review]
Related to https://github.com/chrislusf/seaweedfs/pull/1683

These two fixes are not needed. Need to find the original problem and prevent one volume being loaded twice.

File: weed/storage/disk_location.go (hunk 1 above)
Comment (qieqieplus) [reply weed/storage/disk_location.go:213]: explained in #1683

**Comment by qieqieplus** [comment]
Still the other issue I mentioned in this pull request comment has not been addressed.

**Comment by chrislusf** [comment]
Added a fix. How is it now?

**Comment by qieqieplus** [comment]
LGTM

**Comment by chrislusf** [comment]
Thanks!

**Pull request closed without merging**